src/ralph/backends/http/base.py

```diff
@@ -98,19 +99,46 @@ def validate_query(self, query: Union[str, dict, BaseQuery] = None) -> BaseQuery
 
         return query
 
+    @staticmethod
+    async def _gather_with_limited_concurrency(num_tasks: Union[None, int], *tasks):
```

Comment:
Not sure if this method would be used for other HTTP backends, would it?
I would either put it in `utils.py` (seems generic enough) or as a private method of `AsyncLRSHTTP`. WDYT?

Reply:
Good point. I put this in `utils.py` ! This seems like it may be used in other async backends.